client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+      updatedSheets = [{ ...updatedSheets[0], name: defaultSheetName(1) }]
+    }
+
+    // Update client-side state first for immediate UI feedback
+    setSheets(updatedSheets)
+    setSelectedTabId(
+      // Auto-select the next sheet (or last sheet if none)
+      updatedSheets[Math.min(currentSheetIndex, updatedSheets.length - 1)].id
+    )
+    await saveBatchResults(updatedSheets.map(sheetWithoutId))
+  }
+
+  // ---------- Handlers (end) ----------
+
+  return (
+    <Detail>
+      <H4>{batch.name}</H4>
+
+      {/* When editing one of multiple sheets, BatchResultTallySheet renders its own tab bar with
+        a sheet name form input replacing the selected tab, so we hide the true tab bar */}
+      {(!isEditing || sheets.length === 1) && (
+        <Tabs
+          id={batch.name}
+          onChange={(newTabId: string) => {
+            setSelectedTabId(newTabId)
+          }}
+          selectedTabId={selectedTabId}
+        >
+          {tabs.map(tab => (
+            <Tab id={tab.id} key={tab.id}>
+              {tab.name}
+            </Tab>
+          ))}
+          <Tabs.Expander />
+          {sheets.length > 1 && (
+            <Button
+              icon="add"
+              disabled={areResultsFinalized}
+              minimal
+              onClick={addSheet}
+            >
+              Add Sheet
+            </Button>
+          )}
+        </Tabs>
+      )}
+
+      <BatchResultTallySheet
+        addSheet={addSheet}
+        areResultsFinalized={areResultsFinalized}
+        batch={batch}
+        contest={contest}
+        deleteSheet={deleteSheet}
+        isEditing={isEditing}
+        key={selectedTabId}
+        selectedTabId={selectedTabId}
+        setAreChangesUnsaved={setAreChangesUnsaved}
+        setIsEditing={setIsEditing}
+        sheets={sheets}
+        updateSheet={updateCurrentSheet}
+      />
+    </Detail>
+  )
+}
+
+interface IBatchResultTallySheetProps {
+  addSheet: () => Promise<void>
+  areResultsFinalized: boolean
+  batch: IBatch
+  contest: IContest
+  deleteSheet: () => Promise<void>
+  isEditing: boolean
+  selectedTabId: string
+  setAreChangesUnsaved: (areChangesUnsaved: boolean) => void
+  setIsEditing: (isEditing: boolean) => void
+  sheets: IBatchResultTallySheetWithId[]
+  updateSheet: (updatedSheet: IBatchResultTallySheet) => Promise<void>
+
+  // Require a key to ensure that the state within this component resets when a different sheet is
+  // selected
+  key: string // eslint-disable-line react/no-unused-prop-types
+}
+
+const BatchResultTallySheet: React.FC<IBatchResultTallySheetProps> = ({
+  addSheet,
+  areResultsFinalized,
+  batch,
+  contest,
+  deleteSheet,
+  isEditing,
+  selectedTabId,
+  setAreChangesUnsaved,
+  setIsEditing,
+  sheets,
+  updateSheet,
+}) => {
+  const tabs = tabsFromSheets(sheets)
+  const isTotalsSheet = selectedTabId === VOTE_TOTALS_TAB_ID
+  const selectedSheet = isTotalsSheet
+    ? undefined
```

Comment:
I've generally had a standard in Arlo of using `null` for intentionally "none" values rather than `undefined`. But I know that's confusing since we have the opposite standard in VxSuite...

Reply:
I'm okay with prioritizing in-repo consistency rather than cross-repo, for now! https://github.com/votingworks/arlo/pull/1661/commits/87a25e572e818ac82cd3d1296997381eab1e2194